.github/workflows/ci.yml

```diff
@@ -1,25 +1,24 @@
 name: CI
 on:
     pull_request: {}
+permissions: read-all
 jobs:
   build:
     name: build
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-java@v1
+    - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # renovate: tag=v2
+    - uses: actions/setup-java@f0bb91606209742fe3ea40199be2f3ef195ecabf # renovate: tag=v2
       with:
+        distribution: 'temurin'
         java-version: 8
-    - uses: eskatos/gradle-command-action@v1
+    - uses: gradle/gradle-build-action@937999e9cc2425eddc7fd62d1053baf041147db7 # renovate: tag=v2
       name: gradle
       with:
         arguments: build javadoc
-        wrapper-cache-enabled: true
-        dependencies-cache-enabled: true
-        configuration-cache-enabled: true
```

Comment:
Why do you remove all of these caches?
They are applied only for the current PR, so subsequent commits take less time to rebuild the PR.